cli/src/main/scala/com/codacy/analysis/cli/analysis/ToolSelector.scala

```diff
@@ -38,19 +36,16 @@ class ToolSelector(toolRepository: ToolRepository) {
         duplicationToolsEither.map(_.map(_.to[ITool]))
 
       case Some("issues") =>
-        val toolsEither = tools(None, configuration, languages)
+        val toolsEither = tools(configuration, languages)
         toolsEither.map(_.map(_.to[ITool]))
 
-      case Some(_) =>
-        val toolsEither = tools(toolInput, configuration, languages)
-        toolsEither.map(_.map(_.to[ITool]))
+      case Some(toolInput) =>
+        val toolEither = tool(toolInput, languages)
+        toolEither.map(_.map(_.to[ITool]))
     }
   }
 
-  def tools(toolInput: Option[String],
-            configuration: CLIConfiguration.Tool,
-            languages: Set[Language]): Either[CLIError, Set[Tool]] = {
```

Comment:
This function had almost two separate implementations depending if the `toolInput` parameter was `Some` or `None`.
So I split it into two functions, `tools` and `tool`.